SU2_CFD/src/solvers/CSpeciesFlameletSolver.cpp

```diff
@@ -0,0 +1,841 @@
+        }
+
+        solver_container[i_mesh][SPECIES_SOL]->GetNodes()->SetSolution(i_point, scalar_init);
+      }
+
+      solver_container[i_mesh][SPECIES_SOL]->InitiateComms(geometry[i_mesh], config, SOLUTION);
+      solver_container[i_mesh][SPECIES_SOL]->CompleteComms(geometry[i_mesh], config, SOLUTION);
+
+      solver_container[i_mesh][FLOW_SOL]->InitiateComms(geometry[i_mesh], config, SOLUTION);
+      solver_container[i_mesh][FLOW_SOL]->CompleteComms(geometry[i_mesh], config, SOLUTION);
+
+      solver_container[i_mesh][FLOW_SOL]->Preprocessing(geometry[i_mesh], solver_container[i_mesh], config, i_mesh,
+                                                        NO_RK_ITER, RUNTIME_FLOW_SYS, false);
+    }
+
+    /* --- sum up some counters over processes --- */
+    SU2_MPI::Reduce(&n_not_in_domain_local,  &n_not_in_domain_global,  1, MPI_UNSIGNED_LONG, MPI_SUM, MASTER_NODE, SU2_MPI::GetComm());
+    SU2_MPI::Reduce(&n_not_iterated_local,   &n_not_iterated_global,   1, MPI_UNSIGNED_LONG, MPI_SUM, MASTER_NODE, SU2_MPI::GetComm());
+    SU2_MPI::Reduce(&n_points_unburnt_local, &n_points_unburnt_global, 1, MPI_UNSIGNED_LONG, MPI_SUM, MASTER_NODE, SU2_MPI::GetComm());
+    SU2_MPI::Reduce(&n_points_burnt_local,   &n_points_burnt_global,   1, MPI_UNSIGNED_LONG, MPI_SUM, MASTER_NODE, SU2_MPI::GetComm());
+    SU2_MPI::Reduce(&n_points_flame_local,   &n_points_flame_global,   1, MPI_UNSIGNED_LONG, MPI_SUM, MASTER_NODE, SU2_MPI::GetComm());
+
+    if (rank == MASTER_NODE) {
+      cout << endl;
+      cout << " Number of points in unburnt region: " << n_points_unburnt_global << "." << endl;
+      cout << " Number of points in burnt region  : " << n_points_burnt_global   << "." << endl;
+      cout << " Number of points in flame zone    : " << n_points_flame_global << "." << endl;
+    }
+
+    if (rank == MASTER_NODE && (n_not_in_domain_global > 0 || n_not_iterated_global > 0)) cout << endl;
+
+
+    if (rank == MASTER_NODE && n_not_in_domain_global > 0)
+      cout << " !!! Initial condition: Number of points outside of table domain: " << n_not_in_domain_global << " !!!" << endl;
+
+    if (rank == MASTER_NODE && n_not_iterated_global > 0)
+      cout << " !!! Initial condition: Number of points in which enthalpy could not be iterated: " << n_not_iterated_global
+           << " !!!" << endl;
+
+    if (rank == MASTER_NODE && (n_not_in_domain_global > 0 || n_not_iterated_global > 0)) cout << endl;
+    delete [] scalar_init;
+  }
+}
+
+void CSpeciesFlameletSolver::SetPreconditioner(CGeometry* geometry, CSolver** solver_container, CConfig* config) {
+  unsigned short iVar;
+  unsigned long iPoint, total_index;
+
+  su2double BetaInc2, Density, dRhodT, dRhodC, Temperature, Delta;
+
+  const bool variable_density = (config->GetKind_DensityModel() == INC_DENSITYMODEL::VARIABLE);
+  const bool implicit = (config->GetKind_TimeIntScheme_Flow() == EULER_IMPLICIT);
+
+  for (iPoint = 0; iPoint < nPointDomain; iPoint++) {
```

Comment:
Easy OpenMP

Reply:
@pcarruscag only this?

> SU2_OMP_FOR_STAT(omp_chunk_size)
>    ...
>  END_SU2_OMP_FOR

I think static is OK for this?